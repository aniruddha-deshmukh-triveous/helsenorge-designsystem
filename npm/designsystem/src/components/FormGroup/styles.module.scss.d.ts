export type Styles = {
  'form-group': string;
  'form-group__legend': string;
  'form-group__legend--bigform': string;
  'form-group__legend--on-dark': string;
  'form-group-wrapper': string;
  'form-group-wrapper__errors': string;
  'form-group-wrapper__title': string;
  'form-group-wrapper__title--bigform': string;
  'form-group-wrapper__title--on-dark': string;
  'form-group-wrapper--bigform': string;
  'form-group-wrapper--invalid': string;
  'form-group-wrapper--on-blueberry': string;
  'form-group-wrapper--on-dark': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
