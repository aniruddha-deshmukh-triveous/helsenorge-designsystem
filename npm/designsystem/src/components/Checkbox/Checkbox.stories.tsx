import React from 'react';
import { boolean, select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { FormMode, FormVariant } from '../../constants';
import Checkbox from './Checkbox';
import { mapToBackgoundColor } from '../../../.storybook/StoryBackground';

const stories = storiesOf('Checkbox', module);
stories.addDecorator(withKnobs);

stories.add('Default', () => {
  const modes = select('background', FormMode, FormMode.onwhite);

  return (
    <div
      style={{
        width: '40rem',
        background: mapToBackgoundColor(modes),
        padding: '1rem',
      }}
    >
      <Checkbox
        label={'Jeg er frisk!'}
        checked={boolean('Checked', false)}
        disabled={boolean('Disabled', false)}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
      />
    </div>
  );
});
