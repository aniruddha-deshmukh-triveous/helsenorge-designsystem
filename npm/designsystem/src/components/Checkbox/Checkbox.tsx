import React, { useState } from 'react';

import classNames from 'classnames';

import Check from '../Icons/Check';

import checkboxStyles from './styles.module.scss';

interface CheckboxProps {
  /** Adds custom classes to the element. */
  className?: string;
  /** Om checkbox er checked */
  checked?: boolean;
  /** Teksten som st√•r ved siden av checkbox */
  label: string;
}

export const Checkbox = React.forwardRef((props: CheckboxProps, ref: React.Ref<HTMLUListElement>) => {
  const { className = '', label = '' } = props;
  const [checked, setChecked] = useState(props.checked);

  const checkboxClasses = classNames(checkboxStyles.checkbox, className);

  return (
    <label className={checkboxStyles['checkbox-wrapper']}>
      <input
        className={checkboxClasses}
        type={'checkbox'}
        checked={checked}
        onClick={() => {
          setChecked(!checked);
        }}
      />
      {label}
    </label>
  );
});

export default Checkbox;
