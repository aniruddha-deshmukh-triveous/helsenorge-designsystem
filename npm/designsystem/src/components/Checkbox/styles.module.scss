@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/font-settings.scss';

.checkbox-errors {
  font-size: $font-size-sm;
  font-weight: 600;
  color: $cherry500;
  margin-top: 0;
}

.checkbox-wrapper {
  margin: getSpacer(s) 0;

  &--with-error {
    padding: getSpacer(s) getSpacer(2xs) getSpacer(m);
    background-color: $cherry100;
    border-left: 0.25rem solid $cherry500;

    @media (min-width: map-get($grid-breakpoints, md)) {
      padding: getSpacer(s) getSpacer(m) getSpacer(l);
    }
  }

  &--with-error#{&}--bigform {
    padding: getSpacer(m) getSpacer(m);
  }

  &--bigform {
    margin: getSpacer(xs) 0 getSpacer(xs) 0;
  }
}

.checkbox-wrapper:first-of-type {
  margin-top: 0;
}

.checkbox-wrapper:last-of-type {
  margin-bottom: 0;
}

.checkbox-label {
  display: flex;
  align-items: center;
  cursor: pointer;
  font-size: $font-size-sm;

  @media (min-width: map-get($grid-breakpoints, md)) {
    font-size: $font-size-md;
  }

  &--on-dark {
    color: $white;
  }

  &--invalid {
    color: $black;
  }

  &--disabled {
    cursor: default;
    color: $neutral400;
  }

  &--bigform {
    padding: getSpacer(s);
    background-color: $neutral50;
  }

  &--bigform#{&}--on-blueberry {
    background-color: $blueberry50;
  }

  &--bigform#{&}--on-dark {
    background-color: $blueberry800;
  }

  &--bigform#{&}--invalid {
    background-color: $white;
  }

  @media (min-width: map-get($grid-breakpoints, md)) {
    &--bigform {
      padding: 1.68rem getSpacer(l);
    }
  }
}

.checkbox {
  width: 0px;
  margin: 0;

  -moz-appearance: none;
  -webkit-appearance: none;
  -o-appearance: none;

  &__icon-wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: getSpacer(4xs) getSpacer(s) getSpacer(4xs) getSpacer(4xs);
    height: 1.625rem;
    width: 1.625rem;
    min-width: auto !important;
    min-height: auto !important;
    color: $neutral700;
    box-shadow: 0 0 0 getSpacer(4xs);

    :hover > & {
      box-shadow: 0 0 0 getSpacer(3xs);
      background-color: $neutral100;
    }

    :active > & {
      box-shadow: 0 0 0 getSpacer(3xs);
      color: $black;
    }

    &--checked {
      color: $blueberry500;

      :hover > & {
        background-color: $blueberry50;
      }
    }

    &--on-blueberry {
      color: $blueberry500;

      :hover > & {
        background-color: $blueberry100;
      }
    }

    &--on-dark {
      color: $neutral200;

      :hover > & {
        background-color: $inverted-hover;
      }

      &.checkbox__icon-wrapper--checked {
        color: $blueberry200;
      }

      :active > & {
        color: $white;
      }
    }

    &--invalid {
      color: $cherry500;

      :hover > & {
        background-color: $cherry100;
      }

      &.checkbox__icon-wrapper--checked {
        color: $cherry500;
      }

      :active > & {
        color: $black;
      }
    }

    &--disabled {
      color: $neutral400;
      &.checkbox__icon-wrapper--checked {
        color: $neutral400;
      }
      :hover > & {
        box-shadow: 0 0 0 getSpacer(4xs);
        background-color: transparent;
      }

      :active > & {
        color: $neutral400;
      }
    }
  }

  &__icon {
    position: relative;
    top: 0.1rem;
    right: 0.05rem;
  }

  &:focus {
    & + .checkbox__icon-wrapper {
      box-shadow: 0 0 0 getSpacer(3xs);
      color: $black;

      &--on-dark {
        color: $white;
      }

      &--invalid {
        color: $black;
      }
    }
  }
}
