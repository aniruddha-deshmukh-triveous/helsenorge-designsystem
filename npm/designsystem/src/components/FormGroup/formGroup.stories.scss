.error-wrapper-class--no-bottom-margin {
  padding-bottom: 0;
  border: 2px dashed black;
}
.error-wrapper-class {
  border: 2px dashed black;
}
