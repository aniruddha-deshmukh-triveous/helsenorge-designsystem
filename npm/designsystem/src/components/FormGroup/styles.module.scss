@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/font-settings.scss';

.form-group-wrapper {
  &__title {
    margin-bottom: getSpacer(l) !important;

    &--on-dark {
      color: $white;
    }
  }

  &__errors {
    font-size: $font-size-sm;
    font-weight: 600;
    color: $cherry500;
    margin-top: 0;
  }
}

.form-group-wrapper:first-of-type {
  padding-top: getSpacer(xl);
}

.form-group-wrapper:last-of-type {
  padding-bottom: getSpacer(xl);
}

.form-group {
  padding: getSpacer(s) 0 getSpacer(m);

  @media (min-width: map-get($grid-breakpoints, md)) {
    padding: getSpacer(s) 0 getSpacer(l);
  }

  &--invalid {
    margin-left: -(getSpacer(m) + 0.25rem);
    padding-left: getSpacer(m);
    background-color: $cherry50;
    border-left: 0.25rem solid $cherry500;
  }
}

.field-set {
  border: none;
  padding: 0;
  margin: 0;

  &__legend {
    font-size: $font-size-md;
    font-weight: 600;
    margin-bottom: getSpacer(m);

    &--on-dark {
      color: $white;
    }
  }
}
