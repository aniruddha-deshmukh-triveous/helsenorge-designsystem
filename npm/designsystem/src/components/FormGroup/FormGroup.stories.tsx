import React from 'react';
import { select, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import Checkbox from '../Checkbox/Checkbox';
import RadioButton from '../RadioButton/RadioButton';
import Input from '../Input/Input';
import FormGroup from './FormGroup';
import { FormMode, FormVariant } from '../../constants';
import Coronavirus from '../Icons/Coronavirus';
import { mapToBackgoundColor } from '../../../.storybook/StoryBackground';

const stories = storiesOf('FormGroup', module);
stories.addDecorator(withKnobs);

stories.add('Checkbox', () => {
  const modes = select('background', FormMode, FormMode.onwhite);

  return (
    <div
      style={{
        width: '90vw',
        background: mapToBackgoundColor(modes),
      }}
    >
      <FormGroup
        title={text('Title', 'Gruppe tittel')}
        legend={text('Legend 1', 'Første gruppe')}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
      >
        <Checkbox inputId={'Checkbox1'} label={'Checkbox 1'} />
        <Checkbox inputId={'Checkbox2'} label={'Checkbox 2'} />
        <Checkbox inputId={'Checkbox3'} label={'Checkbox 3'} />
      </FormGroup>
      <FormGroup legend={text('Legend 2', 'Andre gruppe')} mode={modes} variant={select('Variant', FormVariant, FormVariant.normal)}>
        <Checkbox label={'Checkbox 4'} />
        <Checkbox label={'Checkbox 5'} />
        <Checkbox label={'Checkbox 6'} />
      </FormGroup>
    </div>
  );
});

stories.add('RadioButton', () => {
  const modes = select('background', FormMode, FormMode.onwhite);

  return (
    <div
      style={{
        width: '90vw',
        background: mapToBackgoundColor(modes),
      }}
    >
      <FormGroup
        title={text('Title', 'Gruppe tittel')}
        legend={text('Legend 1', 'Første gruppe')}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
        name={'radiogroup1'}
      >
        <RadioButton inputId={'RadioButton1'} label={'RadioButton 1'} />
        <RadioButton inputId={'RadioButton2'} label={'RadioButton 2'} />
        <RadioButton inputId={'RadioButton3'} label={'RadioButton 3'} />
      </FormGroup>
      <FormGroup
        legend={text('Legend 2', 'Andre gruppe')}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
        name={'radiogroup2'}
      >
        <RadioButton inputId={'RadioButton4'} label={'RadioButton 4'} />
        <RadioButton inputId={'RadioButton5'} label={'RadioButton 5'} />
        <RadioButton inputId={'RadioButton6'} label={'RadioButton 6'} />
      </FormGroup>
    </div>
  );
});

stories.add('Input', () => {
  const modes = select('background', FormMode, FormMode.onwhite);

  return (
    <div
      style={{
        width: '90vw',
        background: mapToBackgoundColor(modes),
      }}
    >
      <FormGroup
        title={text('Title', 'Gruppe tittel')}
        legend={text('Legend 1', 'Første gruppe')}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
      >
        <Input label={'Input 1'} />
        <Input icon={Coronavirus} label={'Input 2'} />
        <Input icon={Coronavirus} iconRight label={'Input 3'} />
      </FormGroup>
      <FormGroup
        legend={text('Legend 2', 'Andre gruppe')}
        mode={modes}
        variant={select('Variant', FormVariant, FormVariant.normal)}
        name={'radiogroup2'}
      >
        <Input label={'Input 4'} />
        <Input label={'Input 5'} />
        <Input label={'Input 6'} />
      </FormGroup>
    </div>
  );
});
