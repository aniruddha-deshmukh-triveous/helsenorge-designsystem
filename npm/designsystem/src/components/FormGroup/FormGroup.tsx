import React from 'react';

import classNames from 'classnames';

import formGroupStyles from './styles.module.scss';
import Checkbox, { CheckboxProps } from '../Checkbox/Checkbox';
import { AnalyticsId, FormMode, FormVariant } from '../../constants';
import RadioButton, { RadioButtonProps } from '../RadioButton/RadioButton';
import Input, { InputProps } from '../Input/Input';
import Title from '../Title';

export interface FormGroupProps {
  /** title for the the fieldset */
  title?: string;
  /** text placed in the legend tag of the fieldset */
  legend?: string;
  /** Items in the FormGroup component */
  children?: React.ReactNode;
  /** Adds custom classes to the element. */
  className?: string;
  /** Changes the visuals of the formgroup */
  mode?: keyof typeof FormMode;
  /** Changes the visuals of the formgroup */
  variant?: keyof typeof FormVariant;
  /** Error message */
  error?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
  /** Unique identifyer for the child input tags */
  name?: string;
}

export const FormGroup = React.forwardRef((props: FormGroupProps, ref: React.ForwardedRef<HTMLElement>) => {
  const { className, mode = FormMode.onwhite, variant = FormVariant.normal, error, name } = props;
  const onDark = mode === FormMode.ondark;
  const bigform = variant === FormVariant.bigform;
  const formGroupWrapperClasses = classNames(
    formGroupStyles['form-group-wrapper'],
    {
      [formGroupStyles['form-group-wrapper--on-dark']]: onDark,
      [formGroupStyles['form-group-wrapper--invalid']]: error,
    },
    className
  );
  const titleClasses = classNames(formGroupStyles['form-group-wrapper__title'], {
    [formGroupStyles['form-group-wrapper__title--on-dark']]: onDark && !error,
    [formGroupStyles['form-group-wrapper__title--bigform']]: bigform,
  });
  const formGroupClasses = classNames(formGroupStyles['form-group'], {
    [formGroupStyles['form-group--invalid']]: error,
  });
  const errorStyles = classNames(formGroupStyles['form-group-wrapper__errors'], {
    [formGroupStyles['form-group-wrapper__errors--bigform']]: bigform,
  });
  const legendClasses = classNames(formGroupStyles['field-set__legend'], {
    [formGroupStyles['field-set__legend--on-dark']]: onDark && !error,
    [formGroupStyles['field-set__legend--bigform']]: bigform,
  });

  return (
    <div data-testid={props.testId} data-analyticsid={AnalyticsId.FormGroup} className={formGroupWrapperClasses}>
      {props.title && (
        <Title className={titleClasses} htmlMarkup={'h4'} appearance={'title4'} margin={{ marginTop: 0, marginBottom: 2 }}>
          {props.title}
        </Title>
      )}
      <div className={formGroupClasses}>
        {error && <p className={errorStyles}>{error}</p>}
        <fieldset name={props.title} className={formGroupStyles['field-set']}>
          {props.legend && <legend className={legendClasses}>{props.legend}</legend>}
          {React.Children.map(props.children, (child: React.ReactNode) => {
            if ((child as React.ReactElement<CheckboxProps>).type === Checkbox) {
              let checkbox = (child as React.ReactElement<CheckboxProps>).type === Checkbox;
              return React.cloneElement(child as React.ReactElement<CheckboxProps>, {
                name: name ?? checkbox.valueOf.name,
                mode,
                variant,
                error: !!error,
              });
            } else if ((child as React.ReactElement<RadioButtonProps>).type === RadioButton) {
              let radioButton = (child as React.ReactElement<RadioButtonProps>).type === RadioButton;
              return React.cloneElement(child as React.ReactElement<RadioButtonProps>, {
                name: name ?? radioButton.valueOf.name,
                mode,
                variant,
                error: !!error,
              });
            } else if ((child as React.ReactElement<InputProps>).type === Input) {
              let input = (child as React.ReactElement<InputProps>).type === Input;
              return React.cloneElement(child as React.ReactElement<InputProps>, {
                name: name ?? input.valueOf.name,
                mode,
                variant,
                error: !!error,
              });
            }
            return child;
          })}
        </fieldset>
      </div>
    </div>
  );
});

export default FormGroup;
