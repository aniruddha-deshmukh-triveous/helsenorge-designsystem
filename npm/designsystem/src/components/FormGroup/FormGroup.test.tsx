import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Checkbox from '../Checkbox/Checkbox';
import FormGroup, { FormGroupModes, FormGroupVariants } from './FormGroup';

describe('Gitt at FormGroup skal vises', (): void => {
  describe('Når FormGroup rendres', (): void => {
    test('Så vises FormGroup', (): void => {
      const { container } = render(
        <FormGroup title={'One amazing title'} legend={'Check out these checkboxes!'}>
          <Checkbox inputid={'Checkbox1'} label={'Checkbox 1'} />
          <Checkbox inputid={'Checkbox2'} label={'Checkbox 2'} />
          <Checkbox inputid={'Checkbox3'} label={'Checkbox 3'} />
        </FormGroup>
      );

      expect(container).toMatchSnapshot();

      const title = screen.getByText('One amazing title');
      expect(title).toBeVisible();
      expect(title.className).toBe('title title--title4 form-group-wrapper__title');

      const legend = screen.getByText('Check out these checkboxes!');
      expect(legend).toBeVisible();
      expect(legend.className).toBe('form-group__legend');

      const wrapper = title.parentElement;
      expect(wrapper).toBeVisible();
      expect(wrapper.className).toBe('form-group-wrapper');
    });
  });

  describe('Når mode settes', (): void => {
    test('Så er stylingen satt riktig på FormGroup', (): void => {
      render(
        <FormGroup title={'One amazing title'} legend={'Check out these checkboxes!'} mode={FormGroupModes.onBlueberry}>
          <Checkbox inputid={'Checkbox1'} label={'Checkbox 1'} />
          <Checkbox inputid={'Checkbox2'} label={'Checkbox 2'} />
          <Checkbox inputid={'Checkbox3'} label={'Checkbox 3'} />
        </FormGroup>
      );

      const wrapper = screen.getByText('One amazing title').parentElement;
      expect(wrapper).toBeVisible();
      expect(wrapper.className).toBe('form-group-wrapper form-group-wrapper--on-blueberry');
    });
  });

  describe('Når variant settes', (): void => {
    test('Så er stylingen satt riktig på FormGroup', (): void => {
      render(
        <FormGroup title={'One amazing title'} legend={'Check out these checkboxes!'} variant={FormGroupVariants.bigform}>
          <Checkbox inputid={'Checkbox1'} label={'Checkbox 1'} />
          <Checkbox inputid={'Checkbox2'} label={'Checkbox 2'} />
          <Checkbox inputid={'Checkbox3'} label={'Checkbox 3'} />
        </FormGroup>
      );

      const wrapper = screen.getByText('One amazing title').parentElement;
      expect(wrapper).toBeVisible();
      expect(wrapper.className).toBe('form-group-wrapper form-group-wrapper--bigform');
    });
  });

  describe('Når FormGroup har children', (): void => {
    test('Så rendres de', (): void => {
      render(
        <FormGroup title={'One amazing title'} legend={'Check out these checkboxes!'}>
          <Checkbox inputid={'Checkbox1'} label={'Checkbox 1'} />
          <Checkbox inputid={'Checkbox2'} label={'Checkbox 2'} />
          <Checkbox inputid={'Checkbox3'} label={'Checkbox 3'} />
        </FormGroup>
      );

      const checkboxArray = screen.getAllByRole('checkbox');
      expect(checkboxArray.length).toBe(3);
    });
  });

  describe('Når FormGroup får error prop satt', (): void => {
    test('Så rendres error melding og styling', (): void => {
      render(
        <FormGroup error={'error error!'} title={'One amazing title'} legend={'Check out these checkboxes!'}>
          <Checkbox inputid={'Checkbox1'} label={'Checkbox 1'} />
          <Checkbox inputid={'Checkbox2'} label={'Checkbox 2'} />
          <Checkbox inputid={'Checkbox3'} label={'Checkbox 3'} />
        </FormGroup>
      );

      const error = screen.getByText('error error!');
      expect(error).toBeVisible();

      const formGroup = screen.getByRole('group').parentElement;
      expect(formGroup.className).toBe('form-group-wrapper form-group-wrapper--invalid');
    });
  });
});
