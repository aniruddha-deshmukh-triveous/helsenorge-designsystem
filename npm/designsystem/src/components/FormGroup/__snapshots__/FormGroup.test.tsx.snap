// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at FormGroup skal vises Når FormGroup rendres Så vises FormGroup 1`] = `
<div>
  <div
    class="form-group-wrapper"
  >
    <h4
      class="title title--title4 form-group-wrapper__title"
      style="margin-top: 0rem; margin-bottom: 0rem;"
    >
      One amazing title
    </h4>
    <div
      class="form-group"
    >
      <fieldset
        class="field-set"
        name="One amazing title"
      >
        <legend
          class="field-set__legend"
        >
          Check out these checkboxes!
        </legend>
        <div
          class="checkbox-wrapper"
        >
          <label
            class="checkbox-label"
            for="Checkbox1"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="checkbox"
              id="Checkbox1"
              name="Checkbox1"
              type="checkbox"
              value="Checkbox 1"
            />
            <span
              class="checkbox__icon-wrapper"
            />
            Checkbox 1
          </label>
        </div>
        <div
          class="checkbox-wrapper"
        >
          <label
            class="checkbox-label"
            for="Checkbox2"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="checkbox"
              id="Checkbox2"
              name="Checkbox2"
              type="checkbox"
              value="Checkbox 2"
            />
            <span
              class="checkbox__icon-wrapper"
            />
            Checkbox 2
          </label>
        </div>
        <div
          class="checkbox-wrapper"
        >
          <label
            class="checkbox-label"
            for="Checkbox3"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="checkbox"
              id="Checkbox3"
              name="Checkbox3"
              type="checkbox"
              value="Checkbox 3"
            />
            <span
              class="checkbox__icon-wrapper"
            />
            Checkbox 3
          </label>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;
